test/functional/tools/system/pip_manager_test.py

```diff
@@ -0,0 +1,123 @@
+    _create_py_hello_world(pip_package_folder)
+    pip_package_folder = pip_package_folder.replace('\\', '/')
+
+    conanfile_pip = textwrap.dedent(f"""
+        from conan import ConanFile
+        from conan.tools.system.pip_manager import Pip
+        from conan.tools.files import copy
+        from conan.tools.layout import basic_layout
+        import platform
+        import os
+
+
+        class PipPackage(ConanFile):
+            name = "pip_hello_test"
+            version = "0.1"
+
+            def layout(self):
+                basic_layout(self)
+
+            def build_requirements(self):
+                Pip(self).install(["{pip_package_folder}"])
+
+            def generate(self):
+                Pip(self).generate()
+
+            def build(self):
+                self.run("hello-world")
+        """)
+
+    client = TestClient(path_with_spaces=False)  # FIXME: the python shebang inside vitual env packages fails when using path_with_spaces
+    client.save({"pip/conanfile.py": conanfile_pip})
+    client.run("build pip/conanfile.py")
+
+    assert "RUN: hello-world" in client.out
+    assert "Hello Test World!" in client.out
+
+
+def test_create_pip_manager():
+
+    pip_package_folder = temp_folder(path_with_spaces=False)
+    _create_py_hello_world(pip_package_folder)
+    pip_package_folder = pip_package_folder.replace('\\', '/')
+
+    conanfile_pip = textwrap.dedent(f"""
+        from conan import ConanFile
+        from conan.tools.system.pip_manager import Pip
+        from conan.tools.files import copy
+        from conan.tools.layout import basic_layout
+        import platform
+        import os
+
+
+        class PipPackage(ConanFile):
+            name = "pip_hello_test"
+            version = "0.1"
+
+            def layout(self):
+                basic_layout(self)
+
+            def package(self):
+                Pip(self, self.package_folder).install(["{pip_package_folder}"])
+
+            def finalize(self):
+                copy(self, "*", src=self.immutable_package_folder, dst=self.package_folder)
+
+            def package_info(self):
+                self.cpp_info.includedirs = []
+                self.cpp_info.libdirs = []
+                python_root = os.path.join(self.package_folder, "pip_venv_pip_hello_test", "Scripts" if platform.system() == "Windows" else "bin")
+                self.buildenv_info.prepend_path("PATH", python_root)
+                self.runenv_info.prepend_path("PATH", python_root)
+        """)
+
+    conanfile = textwrap.dedent("""
+        from conan import ConanFile
+
+
+        class Recipe(ConanFile):
+            name = "pip_test"
+            version = "0.1"
+
+            def requirements(self):
+                self.requires("pip_hello_test/0.1")
+
+            def build(self):
+                self.run("hello-world")
+        """)
+
+    client = TestClient(path_with_spaces=False)  # FIXME: the python shebang inside vitual env packages fails when using path_with_spaces
```

Comment:
Can we add a check and raise an error in ``Pip`` for this use case? It will help to early detect it for users

Reply:
The problem is that it does work in some cases. For example, it doesn't give any errors on my macos, but tests fail on the testing machines. I don't know if raise a warning could be enough, although many users will ignore this message